diff --git a/app/streaming/session.cpp b/app/streaming/session.cpp
index 3e4ec71..64f902f 100644
--- a/app/streaming/session.cpp
+++ b/app/streaming/session.cpp
@@ -1510,7 +1510,8 @@ bool Session::startConnectionAsync()
 
     try {
         NvHTTP http(m_Computer);
-        http.startApp(m_Computer->currentGameId != 0 ? "resume" : "launch",
+        try { http.quitApp(); } catch ( ... ) { };
+        http.startApp("launch",
                       m_Computer->isNvidiaServerSoftware,
                       m_App.id, &m_StreamConfig,
                       enableGameOptimizations,
