diff --git a/vuinputd/src/actions/runtime_data.rs b/vuinputd/src/actions/runtime_data.rs
index 8c5903e..0871467 100644
--- a/vuinputd/src/actions/runtime_data.rs
+++ b/vuinputd/src/actions/runtime_data.rs
@@ -54,7 +54,14 @@ pub fn write_udev_data(content: &str, major: u64, minor: u64) -> io::Result<()>
     }
 
     let path = format!("/run/udev/data/c{}:{}", major, minor);
-    let mut file = File::create(&path)?;
+    let mut file = match File::create(&path) {
+    	Ok(path) => path,
+   		Err(err) => {
+     		let files = fs::read_dir("/run/udev")?;
+       	let list = files.collect::<Vec<_>>();
+     		panic!("{} {path} {err}", &format!("{list:?}"));
+     	},
+    };
     file.write_all(cleaned.as_bytes())?;
 
     Ok(())
diff --git a/vuinputd/src/process_tools/mod.rs b/vuinputd/src/process_tools/mod.rs
index f57e045..fa72917 100644
--- a/vuinputd/src/process_tools/mod.rs
+++ b/vuinputd/src/process_tools/mod.rs
@@ -259,12 +259,18 @@ pub fn start_action(action: Action, ns: &RequestingProcess) -> anyhow::Result<u3
 
     let child = unsafe {
         Command::new("/proc/self/exe")
-            .args([
+            .args(match action {
+            	Action::MknodDevice { path: _, major: _, minor: _ } => vec![
+                "--action",
+                action_json.as_str(),
+              ],
+              _ => vec![
                 "--action",
                 action_json.as_str(),
                 "--target-namespace",
                 ns.nsroot.as_str(),
-            ])
+              ]
+            })
             .pre_exec(|| {
                 // Last resort, if the parent just is killed.
                 libc::prctl(libc::PR_SET_PDEATHSIG, libc::SIGKILL);
