diff --git a/zotify/collections.py b/zotify/collections.py
index a41e5fb..32d1f01 100644
--- a/zotify/collections.py
+++ b/zotify/collections.py
@@ -177,6 +177,7 @@ class Collection:
                 filename = Path(self.playables[0].output_template).name
                 filename = filename.replace("{episode_number}", "*")
                 filename = filename.replace("{track_number}", "*")
+                filename = filename.replace("?", "_")
                 for meta in metadata[count % API_MAX_REQUEST_LIMIT]:
                     filename = filename.replace(
                         "{" + meta.name + "}", fix_filename(meta.string)
