diff --git a/unix/x0vncserver/PollingScheduler.cxx b/unix/x0vncserver/PollingScheduler.cxx
index 3cd23c34..3f5f50d7 100644
--- a/unix/x0vncserver/PollingScheduler.cxx
+++ b/unix/x0vncserver/PollingScheduler.cxx
@@ -72,7 +72,7 @@ void PollingScheduler::newPass()
 
     // First polling pass: initialize statistics.
     m_initialState = false;
-    m_ratedDuration = 0;
+    m_ratedDuration = 10;
     m_sleeping = 0;
     memset(m_errors, 0, sizeof(m_errors));
     m_errorSum = 0;
@@ -114,7 +114,6 @@ void PollingScheduler::newPass()
     //
 
     // Estimation based on keeping up constant interval.
-    m_ratedDuration = m_interval - m_errorSum / 2;
 
     // Estimations based on keeping up desired CPU load.
     int optimalLoadDuration1 = 0;
@@ -145,16 +144,6 @@ void PollingScheduler::newPass()
 #endif
 
     // Choose final estimation.
-    if (m_ratedDuration < optimalLoadDuration) {
-      m_ratedDuration = optimalLoadDuration;
-    }
-    if (m_ratedDuration < 0) {
-      m_ratedDuration = 0;
-    } else if (m_ratedDuration > 500 && m_interval <= 100) {
-      m_ratedDuration = 500;
-    } else if (m_ratedDuration > 1000) {
-      m_ratedDuration = 1000;
-    }
 
 #ifdef DEBUG
     fprintf(stderr, "<final est %3d>\t", m_ratedDuration);
