diff --git a/third-party/libvirtualhid/src/platform/linux/uhid_backend.cpp b/third-party/libvirtualhid/src/platform/linux/uhid_backend.cpp
index b2e0a2c..bc8d1f9 100644
--- a/third-party/libvirtualhid/src/platform/linux/uhid_backend.cpp
+++ b/third-party/libvirtualhid/src/platform/linux/uhid_backend.cpp
@@ -994,7 +994,34 @@ namespace lvh::detail {
     }
 
     int key_code_to_linux(KeyboardKeyCode key_code) {
-      static constexpr std::array<std::pair<KeyboardKeyCode, int>, 47> special_keys {{
+      static constexpr std::array<std::pair<KeyboardKeyCode, int>, 74> special_keys {{
+        {0xA6, KEY_BACK},
+        {0xA7, KEY_FORWARD},
+        {0xA8, KEY_REFRESH},
+        {0xA9, KEY_STOP},
+        {0xAA, KEY_SEARCH},
+        {0xAB, KEY_BOOKMARKS},
+        {0xAC, KEY_HOMEPAGE},
+        {0xAD, KEY_MUTE},
+        {0xAE, KEY_VOLUMEDOWN},
+        {0xAF, KEY_VOLUMEUP},
+        {0xB0, KEY_NEXTSONG},
+        {0xB1, KEY_PREVIOUSSONG},
+        {0xB2, KEY_STOPCD},
+        {0xB3, KEY_PLAYPAUSE},
+        {0xB4, KEY_EMAIL},
+        {0xB5, KEY_MEDIA},
+        {0xB6, KEY_PROG1},
+        {0xB7, KEY_PROG2},
+        {0x5D, KEY_COMPOSE},
+        {0x18, KEY_MUHENKAN},
+        {0x1A, KEY_HENKAN},
+        {0x1C, KEY_HIRAGANA},
+        {0x1D, KEY_ZENKAKUHANKAKU},
+        {0x1E, KEY_RO},
+        {0x1F, KEY_YEN},
+        {0x97, KEY_BRIGHTNESSDOWN},
+        {0x98, KEY_BRIGHTNESSUP},
         {0x08, KEY_BACKSPACE},
         {0x09, KEY_TAB},
         {0x0D, KEY_ENTER},
